hive_generator/lib/src/type_adapter_generator.dart

```diff
@@ -42,6 +42,16 @@ class TypeAdapterGenerator extends GeneratorForAnnotation<HiveType> {
       void write(BinaryWriter writer, ${cls.name} obj) {
         ${builder.buildWrite()}
       }
+
+      @override
+      int get hashCode => typeId.hashCode;
+
+      @override
+      bool operator ==(Object other) =>
+          identical(this, other) ||
+          other is $adapterName &&
```

Comment:
I would remove this check. If the runtime types are equal, this check is always true.

Reply:
The &&'s will make sure it is not always equal i believe